lib/connection-parameters.js

```diff
@@ -43,12 +42,12 @@ var useSsl = function () {
 
 var ConnectionParameters = function (config) {
   // if a string is passed, it is a raw connection string so we parse it into a config
-  config = typeof config === 'string' ? parse(config) : (config || {})
+  config = typeof config === 'string' ? parse(config)
```

Comment:
This might be a bit too compact. Maybe:

```javascript
config = typeof config === 'string' ? parse(config) : Object.assign({}, config)

if (config.connectionString) {
  Object.assign(config, parse(config.connectionString))
}
```
?